File: vseros/2014-15/okr/string.py
def erase(s, i):
    return s.replace(s[0:i+1], s[0:i], 1)
    
def middle(a, b, c):
    return a + b + c - min(a, b, c) - max(a, b, c)

s = [[input(), []] for i in range(3)]
for pair in s:
    i = 1
    pair[1].append(1)
    while i < len(pair[0]):
        if pair[0][i-1] == pair[0][i]:
            pair[0] = erase(pair[0], i)
            pair[1][i-1] += 1
        else:
            i += 1
            pair[1].append(1)
if s[0][0] == s[1][0] and s[1][0] == s[2][0]:
    for i in range(len(s[0][0])):
        print(s[0][0][i] * middle(s[0][1][i], s[1][1][i], s[2][1][i]), end='')
    print()
else:
    print('IMPOSSIBLE')

File: vseros/2014-15/okr/string.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

int medium(int a, int b, int c) { return a + b + c - min(a, min(b, c)) - max(a, max(b, c)); }

int main()
{
	const int n = 3;
	string s[3];
	vector<int> v[3];
	for (int i = 0; i < n; ++i)
	{
		cin >> s[i];
		v[i].push_back(1);
		for (size_t j = 1; j < s[i].length();)
		{
			if (s[i][j] == s[i][j - 1])
			{
				++v[i][j - 1];
				s[i].erase(j, 1);
			}
			else
			{
				v[i].push_back(1);
				++j;
			}
		}
	}
	if (s[0] == s[1] and s[1] == s[2])
	{
		for (size_t i = 0; i < s[0].length(); ++i)
		{
			medium(v[0][i], v[1][i], v[2][i]);
			for (int j = 0; j < medium(v[0][i], v[1][i], v[2][i]); ++j)
				cout << s[0][i];
		}
		cout << endl;
	}
	else
	{
		cout << ("IMPOSSIBLE") << endl;
	}
}
